src/containerapp/azext_containerapp/containerapp_session_code_interpreter_decorator.py

```diff
@@ -0,0 +1,152 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+
+from knack.log import get_logger
+from typing import Any, Dict
+
+from azure.cli.core.commands import AzCliCommand
+from azure.cli.command_modules.containerapp.base_resource import BaseResource
+from azure.cli.command_modules.containerapp._utils import (_ensure_location_allowed, CONTAINER_APPS_RP)
+from azure.cli.core.commands.client_factory import get_subscription_id
+from ._clients import SessionPoolPreviewClient
+
+from ._models import SessionCodeInterpreterPythonExecution as SessionPoolModel
+from ._client_factory import handle_raw_exception
+
+logger = get_logger(__name__)
+
+class SessionCodeInterpreterPreviewDecorator(BaseResource):
+    def __init__(self, cmd: AzCliCommand, client: Any, raw_parameters: Dict, models: str):
+        super().__init__(cmd, client, raw_parameters, models)
+        self.session_code_interpreter_def = SessionPoolModel
+
+    def get_argument_name(self):
+        return self.get_param('name')
+
+    def get_argument_resource_group_name(self):
+        return self.get_param('resource_group_name')
+
+    def get_argument_identifier(self):
+        return self.get_param('identifier')
+
+    def get_argument_code(self):
+        return self.get_param('code')
+
+    def get_argument_timeout_in_seconds(self):
+        return self.get_param('timeout_in_seconds')
+
+    def get_argument_filename(self):
+        return self.get_param('filename')
+
+    def get_argument_filepath(self):
+        return self.get_param('filepath')
+        
+    def get_argument_path(self):
+        return self.get_param('path')
+          
+    def get_argument_sessionpool_location(self):
+        return self.get_param('session_pool_location')    
+
+    def get_sessionpool_endpoint(self):
+        # if the user provides the session pool location get session pool endpoint by
+        # constructing the url endpoint ourselves so command runs faster
+        # otherwise get the session pool endpoint by doing a get on the session pool
+        session_pool_endpoint = None
+        if self.get_argument_sessionpool_location() is not None:
+            session_pool_endpoint_fmt = "https://{}.dynamicsessions.io/subscriptions/{}/resourceGroups/{}/sessionPools/{}"
+            session_pool_endpoint = session_pool_endpoint_fmt.format(
+                self.get_argument_sessionpool_location(),
+                get_subscription_id(self.cmd.cli_ctx),
+                self.get_argument_resource_group_name(),
+                self.get_argument_name())
+        else:
+            sessionpool =  SessionPoolPreviewClient.show(cmd=self.cmd, resource_group_name=self.get_argument_resource_group_name(),name=self.get_argument_name())
```

Comment:
About the `SessionPoolPreviewClient`, you can init it in the `__init__`:
self.session_pool_client = SessionPoolPreviewClient

Reply:
added!